pkg/platform/kube/platform_test.go

```diff
@@ -1658,6 +1661,137 @@ func (suite *FunctionKubePlatformTestSuite) TestEnrichFunctionWithPreemptionSpec
+				NodeSelectorTerms: []v1.NodeSelectorTerm{
+					{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      "node-label-key",
+								Operator: v1.NodeSelectorOpNotIn,
+								Values:   []string{"node-label-value"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	defaultCustomAntiAffinity := &v1.Affinity{
+		NodeAffinity: &v1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+				NodeSelectorTerms: []v1.NodeSelectorTerm{
+					{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{
+								Key:      "node-label-key",
+								Operator: v1.NodeSelectorOpNotIn,
+								Values:   []string{"node-label-value"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	defaultCustomNodeSelector := map[string]string{
+		"node-label-key": "node-label-value",
+		"custom-ns":      "custom-ns-value",
+	}
+	for _, testCase := range []struct {
+		name               string
+		preemptionMode     functionconfig.RunOnPreemptibleNodeMode
+		preemptibleNodes   *platformconfig.PreemptibleNodes
+		customNodeSelector map[string]string
+		customAffinity     *v1.Affinity
+	}{
+		{
+			name:               "allow-affinity",
+			preemptionMode:     functionconfig.RunOnPreemptibleNodesAllow,
+			preemptibleNodes:   defaultPreemptibleNodes,
+			customNodeSelector: defaultCustomNodeSelector,
+			customAffinity:     defaultCustomAffinity,
+		},
+		{
+			name:               "allow-anti-affinity",
+			preemptionMode:     functionconfig.RunOnPreemptibleNodesAllow,
+			preemptibleNodes:   defaultPreemptibleNodes,
+			customNodeSelector: defaultCustomNodeSelector,
+			customAffinity:     defaultCustomAntiAffinity,
+		},
+		{
+			name:               "constrain",
+			preemptionMode:     functionconfig.RunOnPreemptibleNodesConstrain,
+			preemptibleNodes:   defaultPreemptibleNodes,
+			customNodeSelector: defaultCustomNodeSelector,
+		},
+		{
+			name:               "prevent-no-affinity",
+			preemptionMode:     functionconfig.RunOnPreemptibleNodesPrevent,
+			preemptibleNodes:   defaultPreemptibleNodes,
+			customNodeSelector: defaultCustomNodeSelector,
+		},
+		{
+			name:               "prevent-with-affinity",
+			preemptionMode:     functionconfig.RunOnPreemptibleNodesPrevent,
+			preemptibleNodes:   defaultPreemptibleNodes,
+			customNodeSelector: defaultCustomNodeSelector,
+			customAffinity:     defaultCustomAntiAffinity,
+		},
+	} {
+		suite.Run(testCase.name, func() {
+			suite.platform.Config.Kube.PreemptibleNodes = testCase.preemptibleNodes
+			functionName := "some-func"
+
+			functionConfig := functionconfig.NewConfig()
+			functionConfig.Meta.Name = functionName
+			functionConfig.Spec.PreemptionMode = testCase.preemptionMode
+			functionConfig.Spec.NodeSelector = testCase.customNodeSelector
+			functionConfig.Spec.Affinity = testCase.customAffinity.DeepCopy()
+
+			for i := 0; i < 2; i++ {
+				// make sure that nothing was added twice
```

Comment:
Why do we need to verify it? so enrichment won't re-add existing labels/affinity/etc ?

Reply:
@TomerShor this is tested like this in the test above this, so I decided to add here as well
>  so enrichment won't re-add existing labels/affinity/etc 
yes